packages/starknet-snap/src/declareContract.ts

```diff
@@ -16,6 +16,11 @@ export async function declareContract(params: ApiParams) {
     const senderAddress = requestParamsObj.senderAddress;
     const network = getNetworkFromChainId(state, requestParamsObj.chainId);
     const { privateKey } = await getKeysFromAddress(keyDeriver, network, state, senderAddress);
+    
+    if(isUpgradeRequired(network, senderAddress)){
+      showUpgradeRequestModal(wallet);
+      return false;
```

Comment:
perhaps throw an error is better?
in terms of end user exp, we handle by dialog
in terms of the coding handle, return false may not fit to the purpose for the caller to determine it is success or not , throw an error seem more general 

WDYT

Reply:
BTW , test is not pass

Reply:
you have to await for showUpgradeRequestModal